File: src/main/java/com/jss/osiris/modules/invoicing/service/PaymentWayService.java
package com.jss.osiris.modules.invoicing.service;

import java.util.List;

import com.jss.osiris.modules.invoicing.model.PaymentWay;

public interface PaymentWayService {
    public List<PaymentWay> getPaymentWays();

    public PaymentWay getPaymentWay(Integer id);
	
	 public PaymentWay addOrUpdatePaymentWay(PaymentWay paymentWay);
}

File: client/src/app/modules/invoicing/services/payment.way.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { PaymentWay } from '../../invoicing/model/PaymentWay';

@Injectable({
  providedIn: 'root'
})
export class PaymentWayService extends AppRestService<PaymentWay>{

  constructor(http: HttpClient) {
    super(http, "invoicing");
  }

  getPaymentWays() {
    return this.getListCached(new HttpParams(), "payment-ways");
  }

  addOrUpdatePaymentWay(paymentWay: PaymentWay) {
    this.clearListCache(new HttpParams(), "payment-ways");
    return this.addOrUpdate(new HttpParams(), "payment-way", paymentWay, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
